src/dumpulator/modules.py

```diff
@@ -1,32 +1,39 @@
-from typing import Dict, Optional, Type, Union, List, Tuple
+from dataclasses import dataclass, field
+from typing import Dict, Optional, Union, List, Tuple
 
 import pefile
 from .memory import MemoryManager
 
 # TODO: support forwarding to API sets
+@dataclass
 class ModuleExport:
-    def __init__(self, address: int, ordinal: int, name: str, forward: Tuple[str, str] = None):
-        self.address = address
-        self.ordinal = ordinal
-        self.name = name
-        self.forward = forward
+    address: int
+    ordinal: int
+    name: str
+    forward: Optional[Tuple[str, str]] = None
 
+@dataclass
 class Module:
-    def __init__(self, pe: pefile.PE, path: str):
-        self.pe = pe
-        path = path.replace("/", "\\")
-        self.path = path
-        self.name = path.split("\\")[-1]
-        self._exports_by_address: Dict[int, int] = {}
-        self._exports_by_ordinal: Dict[int, int] = {}
-        self._exports_by_name: Dict[str, int] = {}
-        self.exports: List[ModuleExport] = []
+    pe: pefile.PE
+    path: str  # TODO(printup): use pathlib.Path

```

Comment:
I think using `pathlib.Path` might cause some confusion on lunix, no?

Reply:
The good part about `pathlib.Path` is that parsing the path (e.g. extracting the file name, extension, etc.) is built-in; therefore it should work fine on linux platforms.